File: app/src/main/java/de/hype/hypenotify/tools/timers/AlwaysRingingTimer.java
package de.hype.hypenotify.tools.timers;

import de.hype.hypenotify.core.interfaces.MiniCore;

import java.time.Instant;
import java.util.UUID;

/**
 * AlwaysRingingTimer - Rings at a set time, no conditions.
 */
public class AlwaysRingingTimer extends BaseTimer {
    public AlwaysRingingTimer(UUID id, Instant time, String message) {
        super(id, null, time, message);
    }

    @Override
    public boolean wouldRing(MiniCore core) {
        return true;
    }
}

File: app/src/main/java/de/hype/hypenotify/app/tools/timers/AlwaysRingingTimer.kt
package de.hype.hypenotify.app.tools.timers

import de.hype.hypenotify.app.core.interfaces.MiniCore
import java.time.Instant
import java.util.*

/**
 * AlwaysRingingTimer - Rings at a set time, no conditions.
 */
class AlwaysRingingTimer(id: UUID?, time: Instant?, message: String?) : BaseTimer(id, null, time, message) {
    override fun wouldRing(core: MiniCore?): Boolean {
        return true
    }
}
